src/scripts/main.js

```diff
@@ -1,3 +1,442 @@
+const thead = document.querySelector('thead');
+let selectedCell;
+let oldValue;
+let newValue = '';
+const newInput = document.createElement('input');
+
+function addEventTocell() {
+  const table = document.querySelector('tbody');
+
+  for (let i = 0; i < table.querySelectorAll('td').length; i++) {
+    table.querySelectorAll('td')[i].addEventListener('click', changeValue);
+  }
+}
+
+function removeEventTocell() {
+  const table = document.querySelector('tbody');
+
+  for (let i = 0; i < table.querySelectorAll('td').length; i++) {
+    table.querySelectorAll('td')[i].removeEventListener('click', changeValue);
+  }
+}
+
+addEventTocell();
+
+for (let i = 0; i < thead.children[0].children.length; i++) {
+  thead.children[0].children[i].setAttribute('status', 'true');
+}
+
+for (let j = 0; j < sortButton.length; j++) {
+  sortButton[j].addEventListener('click', createHTML);
+}
+
+for (let i = 0; i < tbody.rows.length; i++) {
+  data.push({
+    isActive: false,
+    name: tbody.rows[i].cells[0].innerHTML,
+    position: tbody.rows[i].cells[1].innerHTML,
+    office: tbody.rows[i].cells[2].innerHTML,
+    age: tbody.rows[i].cells[3].innerHTML,
+    money: +tbody.rows[i].cells[4].innerHTML.replace(/\D/g, ''),
+  });
+
+  addActionRows();
+}
+
+function addActionRows() {
+  const table = document.querySelector('tbody');
+
+  for (let i = 0; i < table.rows.length; i++) {
+    table.rows[i].addEventListener('click', setActivRow);
+  }
+}
+
+function setActivRow(e) {
+  const table = document.querySelector('tbody');
+  const rows = [...table.rows];
+
+  for (let i = 0; i < table.rows.length; i++) {
+    if (e.currentTarget !== table.rows[i]) {
+      table.rows[i].classList.remove('active');
+      data[i].isActive = false;
+    }
+  }
+
+  if (!data[rows.indexOf(e.currentTarget)].isActive) {
+    e.currentTarget.classList.add('active');
+    data[rows.indexOf(e.currentTarget)].isActive = true;
+  } else {
+    e.currentTarget.classList.remove('active');
+    data[rows.indexOf(e.currentTarget)].isActive = false;
+  }
+}
+
+function checkActivLink() {
+  const table = document.querySelector('tbody');
+
+  for (let i = 0; i < table.rows.length; i++) {
+    if (data[i].isActive === true) {
+      table.rows[i].classList.add('active');
+    }
+  }
+}
+
+function createHTML(e) {
+  const element = e.currentTarget || thead.querySelectorAll('th')[0];
+
+  tbody.innerHTML = '';
+
+  data.map((item) => {
+    const row = document.createElement('tr');
+
+    row.innerHTML = `
+      <td>${item.name}</td>
+      <td>${item.position}</td>
+      <td>${item.office}</td>
+      <td>${item.age}</td>
+      <td class="money">$${item.money.toLocaleString('en')}</td>
+    `;
+    tbody.append(row);
+  });
+  addEventTocell();
+  filterData(element.innerHTML, element.getAttribute('status'), element);
+}
+
+function filterData(filterName, state, element) {
+  if (filterName === 'Name') {
+    if (state === 'true') {
+      data.sort((a, b) => a.name > b.name ? 1 : -1);
```

Comment:
```suggestion
     const propertyName = filterName.toLowerCase();  // тут буде перетворення Name => name

      data.sort((a, b) => a[propertyName] > [propertyName] ? 1 : -1);
```
а так можна зрупувати багато випадків і зменшити кількість коду